sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -334,14 +348,15 @@ DataSource buildDatasource() throws Exception{
       checkArgument(getQuery() != null, "withQuery() is required");
       checkArgument(getRowMapper() != null, "withRowMapper() is required");
       checkArgument(getCoder() != null, "withCoder() is required");
-      checkArgument(
-          getDataSourceConfiguration() != null, "withDataSourceConfiguration() is required");
+      checkArgument((getDataSourceConfiguration() != null || getDataSourceFactory() != null),
```

Comment:
Please also validate that the user didn't specify both.

Reply:
+1